build.gradle

```diff
@@ -0,0 +1,146 @@
+buildscript {
+    ext {
+        artifactory_contextUrl = 'https://ci-artifactory.corda.r3cev.com/artifactory'
+        //deterministic_idea_sdk = '1.8 (CorDapp)'
+        corda_plugins_version = '4.0.29'
+        corda_release_version = '4.0-SNAPSHOT'
+        kotlin_version = '1.2.60'
+        log4j_version = '2.11.1'
+        junit_version = '4.12'
+        hibernate_version = '5.2.6.Final'
+    }
+
+    repositories {
+        mavenLocal()
+        jcenter()
+        maven { url "$artifactory_contextUrl/corda-releases" }
+    }
+
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath "org.jetbrains.kotlin:kotlin-allopen:$kotlin_version"
+        classpath "org.jetbrains.kotlin:kotlin-noarg:$kotlin_version"
+        classpath "net.corda.plugins:quasar-utils:$corda_plugins_version"
+        classpath "net.corda.plugins:cordapp:$corda_plugins_version"
+        classpath "net.corda.plugins:cordformation:$corda_plugins_version"
+    }
+}
+
+apply plugin: 'java'
+apply plugin: 'net.corda.plugins.cordformation'
+apply plugin: 'net.corda.plugins.cordapp'
+
+dependencies {
+    cordapp project(':flow')
+    cordapp project(':contract')
+    cordaRuntime "net.corda:corda:$corda_release_version"
+    cordaRuntime "net.corda:corda-webserver:$corda_release_version"
+    runtimeOnly "org.apache.logging.log4j:log4j-slf4j-impl:$log4j_version"
+}
+
+jar {
+    enabled = false
+}
+
+allprojects {
+    group 'org.testing'
+    version '1.0-SNAPSHOT'
+
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        jcenter()
+        maven { url "$artifactory_contextUrl/corda-releases" } // cordform-common
+        maven { url "https://jitpack.io" }
+    }
+
+    configurations {
+        all {
+            resolutionStrategy {
+                eachDependency { details ->
+                    switch (details.requested.group) {
+                        case 'org.jetbrains.kotlin':
+                            if (details.requested.name.startsWith("kotlin-")) {
+                                // Force dependencies to use the same version of Kotlin as Corda.
```

Comment:
This is a huge amount of code. We should probably have a Gradle Corda plugin, shouldn't we ...